app/sign.js

```diff
@@ -179,31 +206,39 @@ const handleHalfSignEth = function(recoveryRequest, key, skipConfirm, basecoin)
 }
 
 const handleSignEthereum = function(recoveryRequest, key, skipConfirm) {
+  return signEthTx(recoveryRequest, key, skipConfirm, false);
+};
+
+/**
+ * Signs an Ethereum transaction.
+ * @param recoveryRequest The recovery request object.
+ * @param key The provided private key of the wallet.
+ * @param kipConfirm The boolean value that indicates to whether or not to prompt the user to confirm the transaction.
+ * @param isCoinToken The boolean value that indicates if the transaction is for erc20 or ETH.
+ * @returns The 'hex' value of the signed transaction.
+ */
+const signEthTx = function(recoveryRequest, key, skipConfirm, isCoinToken) {
   const EthTx = require('ethereumjs-tx');
 
   const txHex = getTransactionHexFromRequest(recoveryRequest);
   const transaction = new EthTx(txHex);
-  const decimals = coinDecimals[recoveryRequest.coin];
 
-  const customMessage = recoveryRequest.custom ? recoveryRequest.custom.message : 'None';
   const txData = transaction.data;
+  
   const outputs = [{
     address: '0x' + txData.slice(16, 36).toString('hex'),
-    amount: (new BN(txData.slice(36, 68).toString('hex'), 16)).div(TEN.pow(decimals))
+    amount: new BN(txData.slice(36, 68).toString('hex'), 16)
   }];
 
-  confirmRecovery(recoveryRequest.backupKey, outputs, customMessage, skipConfirm);
-
-  if (!key) {
-    console.log('Please enter the xprv of the wallet for signing: ');
-    key = prompt();
+  // if request is for ETH, need to correct the amount decimals.
```

Comment:
To provide more color here, I *think* that we divide by `decimals` for ETH because we *know* the number of decimals for ETH. We support over 200 ERC20 tokens, which all have varying # of decimals -- that would not be reasonable to keep in a static map like `coinDecimals` -- so we just show the user the full precision and let them calculate the unit-value on their own.